plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/providers/StatemachineClassDeclaration.xtend

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2018 committers of YAKINDU and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     committers of YAKINDU - initial API and implementation
+ */
+package org.yakindu.sct.generator.cpp.providers
+
+import com.google.inject.Inject
+import org.yakindu.sct.generator.c.IGenArtifactConfigurations
+import org.yakindu.sct.generator.cpp.CppNaming
+import org.yakindu.sct.generator.cpp.features.GenmodelEntriesExtension
+import org.yakindu.sct.generator.cpp.files.StatemachineHeader
+import org.yakindu.sct.generator.cpp.submodules.InterfaceFunctions
+import org.yakindu.sct.generator.cpp.submodules.InternalFunctions
+import org.yakindu.sct.generator.cpp.submodules.TimingFunctions
+import org.yakindu.sct.generator.cpp.submodules.TracingFunctions
+import org.yakindu.sct.generator.cpp.submodules.lifecycle.LifecycleFunctions
+import org.yakindu.sct.generator.cpp.templates.ClassDeclaration
+import org.yakindu.sct.model.sexec.ExecutionFlow
+import org.yakindu.sct.model.sexec.extensions.SExecExtensions
+import org.yakindu.sct.model.sexec.transformation.StatechartExtensions
+import org.yakindu.sct.model.sgen.GeneratorEntry
+import org.yakindu.sct.model.sgraph.Statechart
+import org.yakindu.sct.model.stext.stext.InternalScope
+
+import static org.yakindu.sct.generator.c.CGeneratorConstants.*
+
+@GeneratorContribution(StatemachineHeader.HEADER_TARGET)
+class StatemachineClassDeclaration implements ISourceFragment {
+	@Inject protected extension CppNaming
+	@Inject protected extension SExecExtensions
+	@Inject protected extension GenmodelEntriesExtension
+	@Inject protected extension StatechartExtensions
+	
+	@Inject protected GeneratorEntry entry
+	
+	@Inject protected extension InternalFunctions
+	@Inject protected extension InterfaceFunctions
+	@Inject protected extension TimingFunctions
+	@Inject protected extension TracingFunctions
+	@Inject protected extension LifecycleFunctions
+	
+	override get(ExecutionFlow it, IGenArtifactConfigurations artifactConfigs) {
+		generateClass(it, artifactConfigs).generate
+	}
+	
+	def generateClass(ExecutionFlow it, extension IGenArtifactConfigurations artifactConfigs) {
```

Comment:
generateClass is not a name that really matches the intend. 'generate' should be used for text transformations. Here the class definition is assembled or build.